### ISSUE TITLE
package graal crashes in Windows11

### ISSUE THREAD

**Issue by philwalk** [VirtusLab/scala-cli#2743]
**Version(s)**
```
# scala-cli version
Scala CLI version: 1.1.3
Scala version (default): 3.3.1
```
# os
```
Windows 11, Java 17.0.9
```
# java -version
```
openjdk version "17.0.9" 2023-10-17 LTS
OpenJDK Runtime Environment Zulu17.46+19-CA (build 17.0.9+8-LTS)
OpenJDK 64-Bit Server VM Zulu17.46+19-CA (build 17.0.9+8-LTS, mixed mode, sharing)
```

**Describe the bug**
Attempt to package hello.sc with `native-image` fails.

**To Reproduce**
```scala-cli
#!/usr/bin/env -S scala-cli shebang

//> using scala "3.3.1"
//> using dep "com.lihaoyi::os-lib::0.9.3"

import os.Path

main(args)
def main(args: Array[String]): Unit = {
  printf("%s\n", os.Path("/"))
}
```
```sh
# scala-cli --power package /opt/ue/clisrc/hello.sc --graal
Invalid parameter - E:
Error: os.SubprocessException: Result of cmdà: 1

For more details, please see 'C:\opt\ue\clisrc\.scala-build\stacktraces\1708383535-2865679820207472018.log'
```
The logfile:<details>
<pre>
Aliasing C:\Users\user\AppData\Local\Coursier\cache\arc\https\github.com\graalvm\graalvm-ce-builds\releases\download\vm-22.3.1\graalvm-ce-java17-windows-amd64-22.3.1.zip to E:\
Invalid parameter - E:
Exception in thread "main" os.SubprocessException: Result of cmdà: 1

        at os.SubprocessException$.apply(Model.scala:211)
        at os.proc.call(ProcessOps.scala:95)
        at scala.cli.packaging.NativeImage$.maybeWithShorterGraalvmHome(NativeImage.scala:136)
        at scala.cli.packaging.NativeImage$.buildNativeImage$$anonfun$1(NativeImage.scala:264)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.build.internal.ManifestJar$.maybeWithManifestClassPath(ManifestJar.scala:75)
        at scala.cli.packaging.NativeImage$.buildNativeImage(NativeImage.scala:267)
        at scala.cli.commands.package0.Package$.doPackage$$anonfun$1(Package.scala:410)
        at scala.build.EitherCps$Helper.apply(EitherCps.scala:19)
        at scala.cli.commands.package0.Package$.doPackage(Package.scala:523)
        at scala.cli.commands.package0.Package$.runCommand(Package.scala:155)
        at scala.cli.commands.package0.Package$.runCommand(Package.scala:71)
        at scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:368)
        at scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:350)
        at caseapp.core.app.CaseApp.main(CaseApp.scala:157)
        at scala.cli.commands.ScalaCommand.main(ScalaCommand.scala:335)
        at caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:166)
        at scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:125)
        at scala.cli.ScalaCli$.main0(ScalaCli.scala:269)
        at scala.cli.ScalaCli$.main(ScalaCli.scala:108)
        at scala.cli.ScalaCli.main(ScalaCli.scala)
</pre>
</details>

**Expected behaviour**
When I manually issue the `SUBST` command from a CMD session, I get the following results:
```dos
c:\scala-cli>subst E: C:\Users\user\AppData\Local\Coursier\cache\arc\https\github.com\graalvm\graalvm-ce-builds\releases\download\vm-22.3.1\graalvm-ce-java17-windows-amd64-22.3.1.zip
Invalid parameter - E:
```
By contrast, it works for `G:` as shown here:
```dos
c:\scala-cli>subst G: C:\Users\user\AppData\Local\Coursier\cache\arc\https\github.com\graalvm\graalvm-ce-builds\releases\download\vm-22.3.1\graalvm-ce-java17-windows-amd64-22.3.1.zip
```

The following command lists MountedDevices in the Windows registry:
```
$ reg query 'HKEY_LOCAL_MACHINE\SYSTEM\MountedDevices' | grep 'DosDevices' | sed -E -e 's#REG_BINARY.*##'
    \DosDevices\C:
    \DosDevices\D:
    \DosDevices\E:
    \DosDevices\F:
    \DosDevices\H:
```
So apparently the method `availableDriveLetter` is insufficient for determining which drive letters are actually available.

Here's a version that seems to work:
```scala
#!/usr/bin/env -S scala-cli shebang

//> using scala "3.3.1"
//> using dep "com.lihaoyi::os-lib::0.9.3"

import os.*

DriveLetters.main(args)
object DriveLetters {
  def main(args: Array[String]): Unit = {
    printf("%s\n", availableDriveLetter())
  }
  private def availableDriveLetter(): Char = {
    import scala.annotation.tailrec
    @tailrec
    def helper(from: Char): Char =
      if (from > 'Z') sys.error("Cannot find free drive letter")
      else {
        if (unavailableDriveLetters.contains(from)) helper((from + 1).toChar)
        else from
      }

    helper('D')
  }
  lazy val unavailableDriveLetters: String = {
    val queryMountedDosDevices = "reg query HKEY_LOCAL_MACHINE/SYSTEM/MountedDevices".replace('/', '\\')
    val lines = os.proc("cmd", "/c", queryMountedDosDevices).call().out.lines()
    val dosDevices = lines.filter { s =>
      s.contains("DosDevices")
    }.map { s =>
      s.replaceAll(".DosDevices.", "").replaceAll(":.*", "")
    }
    dosDevices.mkString
  }
}
```

**Comment by philwalk** [comment]
Will submit a PR today

**Comment by philwalk** [comment]
This is fixed by PR #2749
